File: 백준/Silver/1002. 터렛/터렛.py
t = int(input())

for i in range(t):
    x1,y1,r1,x2,y2,r2 = map(int, input().split())
    dis = (abs(x1-x2)**2 + abs(y1-y2)**2)**(1/2)
    if dis == 0 and r1 == r2:
        print(-1)
    elif dis > r1+r2 or dis < abs(r1-r2):
        print(0)
    elif dis == r1+r2 or dis == abs(r1-r2):
        print(1)
    else:
        print(2)

File: 백준/Silver/1002. 터렛/터렛.cs

namespace TestingField
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int t = int.Parse(Console.ReadLine());
            for(int test =  0; test < t; ++test)
            {
                int[] input = Console.ReadLine().Split().Select(int.Parse).ToArray();
                int x1 = input[0];
                int y1 = input[1];
                int r1 = input[2];
                int x2 = input[3];
                int y2 = input[4];
                int r2 = input[5];

                //최대 2억
                int dist = (x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2);

                if (x1 == x2 && y1 == y2)
                {
                    if (r1 == r2)
                        Console.WriteLine(-1);
                    else
                        Console.WriteLine(0);
                }
                else if (dist == (r1 + r2) * (r1 + r2) || dist == (r1 - r2) * (r1 - r2))
                    Console.WriteLine(1);
                else if (dist < (r1 + r2) * (r1 + r2) && dist > (r1 - r2) * (r1 - r2))
                    Console.WriteLine(2);
                else
                    Console.WriteLine(0);

            }
        }
    }
}